src/services/dexes/sovryn/SovrynPriceFetcher.ts

```diff
@@ -0,0 +1,87 @@
+import {inject, injectable} from 'inversify';
+import path from 'path';
+import {BigNumber, Contract} from 'ethers';
+import {readFileSync} from 'fs';
+import {fileURLToPath} from 'url';
+import {BaseProvider} from '@ethersproject/providers';
+
+import {FeedFetcherInterface} from 'src/types/fetchers.js';
+import Settings from 'src/types/Settings';
+import {ProviderRepository} from '../../../repositories/ProviderRepository.js';
+import {ChainsIds} from '../../../types/ChainsIds.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+export interface Price {
+  price: BigNumber;
+  success: boolean;
+}
+
+export type PricesResponse = {
+  prices: Price[];
+  timestamp: BigNumber;
+};
+
+export type PairRequest = {
+  base: string;
+  quote: string;
+  amountInDecimals: bigint | number;
```

Comment:
```suggestion
  amountInDecimals: number;
```

this will be `1..20`